src/Middleware.js

```diff
@@ -52,13 +139,15 @@ export function middleware (collection) {
 }
 
 /**
- * @param {(ret: string) => void} callback
+ * @param {(rule: string) => void} callback
  * @return {Middleware}
  */
 export function rulesheet (callback) {
 	return function (element) {
 		if (!element.root)
+			// @ts-ignore
```

Comment:
Not a fan of the ts-ignore comments, considering this commit essentially create a full type-definition we could just shop a `typings.d.ts` definition file with this in it instead of JSDoc comments?

Reply:
> Not a fan of the ts-ignore comments

Plan to remove them, they were only somewhat helpful to me when I've been typing this.

> considering this commit essentially create a full type-definition we could just shop a typings.d.ts definition file with this in it instead of JSDoc comments?

Yeah, we can totally do that - my idea was just that the types were sort-of already here, so I've tweaked those instead of providing a separate set of types. Would you like me to bring back old&simple JSDocs and gather all of those types within `.d.ts` file?